File: snippets/fsharp/VS_Snippets_CLR/DateTime.CompareTo/FS/class1.fs
// <Snippet1>
open System

let theDay = DateTime(DateTime.Today.Year, 7, 28)

try
    let compareValue = theDay.CompareTo DateTime.Today

    if compareValue < 0 then
        printfn $"{theDay:d} is in the past."
    elif compareValue = 0 then
        printfn $"{theDay:d} is today!"
    else // compareValue > 0
        printfn $"{theDay:d} has not come yet."
        
with :? ArgumentException ->
    Console.WriteLine("Value is not a DateTime");

// </Snippet1>

File: snippets/csharp/System/DateTime/CompareTo/class1.cs
using System;

namespace CompareTo
{
    class Class1
    {
        static void Main(string[] args)
        {
            // <Snippet1>
            System.DateTime theDay = new(System.DateTime.Today.Year, 7, 28);
            int compareValue;

            try
            {
                compareValue = theDay.CompareTo(DateTime.Today);
            }
            catch (ArgumentException)
            {
                Console.WriteLine("Value is not a DateTime");
                return;
            }

            if (compareValue < 0)
                System.Console.WriteLine($"{theDay:d} is in the past.");
            else if (compareValue == 0)
                System.Console.WriteLine($"{theDay:d} is today!");
            else // compareValue > 0
                System.Console.WriteLine($"{theDay:d} has not come yet.");
            // </Snippet1>
        }
    }
}
